cranelift/codegen/src/isa/zkasm/abi.rs

```diff
@@ -427,12 +424,15 @@ impl ABIMachineSpec for ZkAsmMachineDeps {
                     RegClass::Vector => unimplemented!("Vector Clobber Saves"),
                 };
                 insts.push(Self::gen_store_stack(
-                    StackAMode::SPOffset(cur_offset, ty),
+                    StackAMode::SPOffset(-cur_offset, ty),
                     real_reg_to_reg(reg.to_reg()),
                     ty,
                 ));
-                cur_offset -= 1
+                cur_offset += 8
```

Comment:
Maintain a negative `cur_offset`, rather than adding up a positive number and negating it on every use (more likely to get it wrong somewhere that way.)